model/src/w3swldmd.F90

```diff
@@ -198,6 +222,26 @@ SUBROUTINE W3SWL4 (A, CG, WN, DAIR, S, D)
 !/
       END SUBROUTINE W3SWL4
 !/ ------------------------------------------------------------------- /
+!
+!> @author H. L. Tolman @date 16-Feb-2012
+!> @param[in] A Action density spectrum (1-D, NTH*NK)
+!> @param[in] CG Group velocities
+!> @param[in] WN Wavenumbers
+!> @param[out] S Source term (1-D, NTH*NK)
+!> @param[out] D Diagonal term of derivative (1-D, NTH*NK)
+!> @returns S,D        
+!> @brief Turbulent dissipation of narrow-banded swell
+!> 
+!> @details
```

Comment:
The details tag is not needed. Nor is the brief tag for subroutines/functions. In the case of a subroutine/function, doxygen will assume the first sentence is the brief description, and the rest is the detailed description, which is a very natural default.

Reply:
Good to know.  So, the brief tag is needed for things other than subroutines/functions, and the details tag is never needed?

Reply:
For the time being I decided to leave the brief and details tags in.  My logic is based on a couple considerations.  The first being that it adds some organization for programmers looking at the code.  It also adds consistency to the headers -- so you don't need to remember which does or doesn't need the tags.  I've added a note to both the subroutine and function templates above about this. I'll be fine either way, so I'm looking forward what other's prefer.